wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java

```diff
@@ -361,7 +366,32 @@ else if (url.toString().startsWith("."))
 					url = url.toString().substring(1);
 				}
 
-				urls.add(url.toString());
+				// The context path and the filter have to be applied to the URL, because otherwise
+				// the resource is not pushed correctly
+				StringBuffer partialUrl = new StringBuffer();
+				String contextPath = WebApplication.get().getServletContext().getContextPath();
+				partialUrl.append(contextPath);
+				if (!contextPath.equals("/"))
+				{
+					partialUrl.append("/");
+				}
+				String filterPath = WebApplication.get().getWicketFilter().getFilterPath();
+				if (filterPath.equals("/"))
+				{
+					filterPath = "";
+				}
+				else if (filterPath.endsWith("/"))
+				{
+					filterPath = filterPath.replaceAll(".$", "");
+				}
+				partialUrl.append(filterPath);
```

Comment:
@martin-g - long time ago we wrote to each other 😄 - I have a question regarding receiving the context path / filter path - is there a method to do so?

Example:

https://127.0.0.1:8443/mywebappcontext/myfilter/wicket/....

I need a better way to get this part:

/mywebappcontext/myfilter/